### ISSUE TITLE
Error with 2.0.13.1 and 2.0.12 versions working together

### ISSUE THREAD

**Issue by Ni-san** [yiisoft/yii2#15428]
I have few versions of app running on one server. When I tried to upgrade one of them to 2.0.13, I got the follow error in other versions:
```
yii\base\ErrorException: Uncaught yii\base\ErrorException: Trying to get property of non-object in /project-path/vendor/yiisoft/yii2/db/ActiveRecord.php:404
Stack trace:
#0 /project-path/vendor/yiisoft/yii2/db/ActiveRecord.php(404): yii\base\ErrorHandler->handleError(8, 'Trying to get p...', '/project-pa...', 404, Array)
#1 /project-path/vendor/yiisoft/yii2/db/ActiveQueryTrait.php(134): yii\db\ActiveRecord::populateRecord(Object(app\models\common\Lang), Array)
#2 /project-path/vendor/yiisoft/yii2/db/ActiveQuery.php(209): yii\db\ActiveQuery->createModels(Array)
#3 /project-path/vendor/yiisoft/yii2/db/ActiveQuery.php(296): yii\db\ActiveQuery->populate(Array)
#4 /project-path/components/query/ExtendedActiveQuery.php(38): yii\db\ActiveQuery->one(NULL)
#5 /project-path/vendor/yiisoft/yii2/db/BaseActiveRecord.php(107): app\components\query\Ext in /project-path/vendor/yiisoft/yii2/db/ActiveRecord.php:404
Stack trace:
#0 [internal function]: yii\base\ErrorHandler->handleFatalError()
#1 {main}
```
As far as I understand, schema cache has different structure in different versions, but stored with same key.

Is there some workaround, like setting custom key for schema?

### What steps will reproduce the problem?
Run two yii2-applications with different versions, shared cache and shared database.
### What is the expected result?
Cache stores same structures with same keys, different structures with different keys.
### What do you get instead?
yii\base\ErrorException: Trying to get property of non-object

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.13.1
| PHP version      | 7.0.15
| Operating system |  FreeBSD 10.3-RELEASE amd64

**Comment by Ni-san** [comment]
Maybe related #15117

**Comment by tehmaestro** [comment]
Hi, I just experienced the same error. Yii 2.0.12 and Yii 2.0.15 . Does the fix from 2.0.16 https://github.com/yiisoft/yii2/commit/ab092f2e2178e2d7ac91f2edc21b3d05825ae9bc work? Thanks.
